### ISSUE TITLE
[Bug] Issue with python 3.12 support

### ISSUE THREAD

**Issue by PrinceJunkie** [aminalaee/uuid-utils#30]
I'm using the devcontainer `mcr.microsoft.com/vscode/devcontainers/python:3.12-bullseye` as an environment.

If I try to install the latest verson of uuid-utils it fails:

```shell
vscode ➜ /dev (master) $ pip install uuid-utils
Defaulting to user installation because normal site-packages is not writeable
Collecting uuid-utils
  Downloading uuid_utils-0.4.0.tar.gz (15 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Checking for Rust toolchain....
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

It works if you change the devcontainer to python:3.11-bullseye

**Comment by aminalaee** [comment]
I need to do a release to publish 3.12 wheels, apparently it's trying to download the `Downloading uuid_utils-0.4.0.tar.gz ` and build it locally, but again complains that:

```
Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
```

I think the case is that if you switch to 3.11 it's not downloading the tar file but instead getting the wheel so you don't need Cargo.

**Comment by aminalaee** [comment]
This is done now.

**Comment by PrinceJunkie** [comment]
@aminalaee 

I'm having the same issue with the new version again:

```shell

Collecting uuid_utils==0.5.0 (from -r requirements-dev.txt (line 18))
  Downloading uuid_utils-0.5.0.tar.gz (15 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Checking for Rust toolchain....
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

```

**Comment by aminalaee** [comment]
I think you should probably check here: https://pypi.org/project/uuid-utils/0.5.0/#files
As your OS might not be one of those which has 3.12 wheel ready.

**Comment by PrinceJunkie** [comment]
@aminalaee 

Could you please explain on what the "wheel readiness" depends?

Because the OS used in the devcontainer had no problems with python 3.11 and I see wheels called `manylinux` for python 3.11 but not for 3.12 😕 

Anyway: Keep up the great work, thanks for everything what you did so far!

**Comment by PrinceJunkie** [comment]
@aminalaee https://github.com/pypa/manylinux/issues/1484 added support for CPython 3.12.  
Could you please add the wheels for the missing distros?

**Comment by opentyler** [comment]
I'm running into this error myself as well, but also thank you for putting this together library.

**Comment by aminalaee** [comment]
Seems like Python 3.12 is still not available, there might be an issue with maturin-action project:
https://github.com/aminalaee/uuid-utils/pull/36

**Comment by opentyler** [comment]
Yeah that makes sense, maturin was what was causing my build to fail last night. That being said, I noticed the maturin version it was trying to download was 0.14.17, which seems weird to be such an old version since newest maturin is version 1.3. Tho it is specified as such in the pyproject.toml ([build-system] requires = ["maturin>=0.14,<0.15"]). Is it possible it's just a maturin version issue? 

Thanks again for responding so quickly and my apologies if the above is a stupid question.

**Comment by opentyler** [comment]
Thanks for the commits!

That pretty well solved it. Then I just had to do 1 more thing to be able to install successfully, which was enable access to the unstable features of the rust uuid crate (it was giving me errors it couldnt access these features), so I solved this by setting the following environment variable: 
        export RUSTFLAGS="--cfg uuid_unstable"

Your commits and then setting that environment variable has enabled me to get uuid_utils to install correctly, so thanks again!

**Comment by aminalaee** [comment]
I will do a release today so the wheels will be available and you won’t need to build locally.

**Comment by aminalaee** [comment]
This should be fixed in 0.6.0 release.

**Comment by PrinceJunkie** [comment]
Did I miss something?

```
Collecting uuid_utils==0.6.0 (from -r requirements-dev.txt (line 18))
  Downloading uuid_utils-0.6.0.tar.gz (16 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Checking for Rust toolchain....
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by PrinceJunkie** [comment]
@aminalaee 

Did I miss something?

```
Collecting uuid_utils==0.6.0 (from -r requirements-dev.txt (line 18))
  Downloading uuid_utils-0.6.0.tar.gz (16 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Checking for Rust toolchain....
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

**Comment by aminalaee** [comment]
what OS is that? they have all the wheels for Python 3.12 now.

**Comment by PrinceJunkie** [comment]
The environment is the devcontainer `mcr.microsoft.com/vscode/devcontainers/python:3.12-bullseye` (or python3.11 respectively).
The OS is Debian Bullseye. Version 0.5.0 worked fine for python 3.11 but 0.60 does not.

**Comment by aminalaee** [comment]
Thank you for flagging this, the wheels were not uploaded completely.
I have confirmed the fix works for Python 3.12, just bump to `0.6.1`
